### PR TITLE
Custom geojson style testing environment for  METEOSCHWEIZ

### PR DESCRIPTION
**Pull request by procrastinatio**
This PR tries to address (or replace) the following https://github.com/geoadmin/mf-chsdi3/pull/2931/files

Basically, METEOSCHWEIZ needs a test environnment to check the validity of its newly developped GeoJSON style file.

See [BGDIINF_SB-2184](https://jira.swisstopo.ch/browse/BGDIINF_SB-2184) Custom config for MeteoSchweiz

To test:

A new custom set of configuration files has been uploaded to: https://mf-chsdi3.dev.bgdi.ch/public_ltmom/configs/

Now, you may call `map.geo.admin.ch` as https://map.geo.admin.ch/?config_url=%2F%2Fmf-chsdi3.dev.bgdi.ch%2Fpublic_ltmom%2Fconfigs&lang=de&topic=meteoschweiz&layers=ch.meteoschweiz.messwerte-neuschnee-1d 

Now, both the **GeoJSON** data file
https://data.geo.admin.ch/ch.meteoschweiz.messwerte-neuschnee-1d/ch.meteoschweiz.messwerte-neuschnee-1d_de.json
and the **style** (`testing`)
https://data.geo.admin.ch/ch.meteoschweiz.messwerte-neuschnee-1d/testing/ch.meteoschweiz.messwerte-neuschnee-1d.json

are hosted on the same AWS S3 bucket.

Normally (https://map.geo.admin.ch/?lang=de&topic=meteoschweiz&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.meteoschweiz.messwerte-neuschnee-1d) only the the **style** file is hosted in `mf-chsdi3`

The **style** file
https://api3.geo.admin.ch/static/vectorStyles/ch.meteoschweiz.messwerte-neuschnee-1d.json

Very custom style (red outlined circles)
https://map.geo.admin.ch/?config_url=%2F%2Fmf-chsdi3.dev.bgdi.ch%2Fpublic_ltmom%2Fconfigs&lang=de&topic=meteoschweiz&layers=ch.meteoschweiz.messwerte-neuschnee-2d

### CODE REVIEW

**Comment by davidoesch** [comment]
Awesome—- let me know as soon as I Can forward a testlink /docu

**Comment by boecklic** [comment]
As discussed: Style file for testing should be hosted on bucket cms.geo.admin.ch in aws-admin account

**Comment by procrastinatio** [comment]
**GeoJson data** (unchangend)

bucket _s3://data.geo.admin.ch_
Url _https://data.geo.admin.ch/ch.meteoschweiz.*_

**Config to custom meteoconfigs** (new)

Use the param: _config_url_ 
S3 bucket: _s3://mf-geoadmin3-dev-dublin/meteoconfigs/_
URL : _https://mf-geoadmin3.dev.bgdi.ch/meteoconfigs/_

**Testing style** (moved)

old _s3://data.geo.admin.ch_
new _s3://cms.geo.admin.ch/ch.meteoschweiz_
url: _https://cms.geo.admin.ch/ch.meteoschweiz/_

**Image for testing** (moved)
old _s3://data.geo.admin.ch/ch.meteoschweiz_
new _s3://cms.geo.admin.ch/ch.meteoschweiz_
URL: _https://data.geo.admin.ch/ch.meteoschweiz_

:note: For the integration, you'll have to copy the images and json files, and update the path.


Testing

https://map.geo.admin.ch/?config_url=//mf-geoadmin3.dev.bgdi.ch/meteoconfigs/

or with a custom meteoschweiz layer:

https://map.geo.admin.ch/?config_url=%2F%2Fmf-geoadmin3.dev.bgdi.ch%2Fmeteoconfigs%2F&lang=fr&topic=meteoschweiz&bgLayer=voidLayer&layers=ch.bafu.gefahren-basiskarte,ch.meteoschweiz.messwerte-wind-boeenspitze-kmh-6h&layers_opacity=0.7,1&catalogNodes=15046,15055,15126,15138,15052

Verify that the Style file and the various images are hosted on _https://cms.geo.admin.ch_

File: METEOSCHWEIZ.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    https://map.geo.admin.ch?config_url=//mf-geoadmin3.dev.bgdi.ch/meteoconfigs
+
+
+Style and resources for testing
+===============================
+
+Testing styles for GeoJson are to be on the location defined in the above mentionned _meteoconfigs_ files, 
+currently:
+
+    _https://cms.geo.admin.ch/ch.meteoschweiz/{bodLayerId}/testing/{bodLayerId}.json
```
Comment (pakb) [inline METEOSCHWEIZ.md:58]:
```suggestion
    _https://cms.geo.admin.ch/ch.meteoschweiz/{bodLayerId}/testing/{bodLayerId}.json_
```

File: METEOSCHWEIZ.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Background information
+======================
+
+Most of the `map.geo.admin.ch` layers, topics and catalog configurations are hold in a few 
+configuration files. Theses files are statified services in `map.geo.admin.ch` deployment to
+the `int` and `prod` environment, but real services on `dev`
+
+The location of these files may be controlled by the parameter `config_url`.
+
+See https://github.com/geoadmin/mf-geoadmin3/pull/4764 poru discussions and implementation
```
Comment (pakb) [inline METEOSCHWEIZ.md:19]:
```suggestion
See https://github.com/geoadmin/mf-geoadmin3/pull/4764 for discussions and implementation
```

File: mk/serviceconfig.mk [hunk 3]
```diff
@@ -28,6 +28,15 @@ configs/: .build-artefacts/last-version \
 	$(foreach topic, $(TOPICS), $(foreach lang, $(LANGS),curl -s --retry 3 --fail -o configs/${lang}/catalog.${topic}.json http:$(API_URL)/rest/services/$(topic)/CatalogServer?lang=$(lang) && jq . configs/${lang}/catalog.${topic}.json 1> /dev/null; ))
 
 
+meteoconfigs/: export METEO_TESTING_STYLE_BASEURL=$(METEO_STYLE_BASEURL)
+meteoconfigs/:  configs/
```
Comment (pakb) [inline mk/serviceconfig.mk:32]: what is achieved by declaring two times the same rule in a row? won't the last rule overwrite the first?

**Review by pakb — APPROVED** [review]

File: METEOSCHWEIZ.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+**Testing style** (moved)
+
+old _s3://data.geo.admin.ch_
+new _s3://cms.geo.admin.ch/ch.meteoschweiz_
+url: _https://cms.geo.admin.ch/ch.meteoschweiz/_
+
+**Image for testing** (moved)
+old _s3://data.geo.admin.ch/ch.meteoschweiz_
+new _s3://cms.geo.admin.ch/ch.meteoschweiz_
+URL: _https://data.geo.admin.ch/ch.meteoschweiz_
```
Comment (boecklic) [inline METEOSCHWEIZ.md:89]: shouldn't this be `URL: https://cms.geo.admin.ch/ch.meteoschweiz` ?

File: mk/serviceconfig.mk [hunk 5]
```diff
@@ -28,6 +28,15 @@ configs/: .build-artefacts/last-version \
...
+	${PYTHON_CMD}  scripts/meteoStyleUrl.py
+
+.PHONY: s3uploadmeteoconfig
+s3uploadmeteoconfig: meteoconfigs/
```
Comment (boecklic) [inline mk/serviceconfig.mk:36]: maybe these targets should somehow go into the repo bgdi-scripts? to have all data deployment related stuff together there?

File: METEOSCHWEIZ.md (hunk 4 above)
Comment (procrastinatio) [reply METEOSCHWEIZ.md:89]: Yes

File: mk/serviceconfig.mk (hunk 5 above)
Comment (procrastinatio) [reply mk/serviceconfig.mk:36]: Don't know. We already have a `s3uploadmeteoconfig` rule and these custom meteo config will never ever be changed before _map.ge.admin.ch_ end of life-

**Pull request merged**